### PR TITLE
improve ledger ux

### PR DESCRIPTION
**Pull request by rolandpo**
Ledger UX improvements from issue #845
The two step process for selecting a Ledger derivation path first and then selecting an address is merged into one modal. If the user has their Ledger connected and the Ethereum app open then 5 addresses from each path get preloaded, the first path to have a non-zero ETH balance in one of their addresses gets displayed, and all 5 addresses have their ETH and aggregate stablecoin USDT, DAI and USDC balances displayed underneath. Paths can still be switched afterwards.

If there is some problem with discovering Ledger at first, then the modal requires a button click to choose the path once the Ledger gets discovered.

On each path, an additional 5 addresses can be displayed, so 3 x 10 choices overall.

Hopefully it doesn't take to long to render initially, might be possible to do some optimisation trade-offs etc otherwise.

ledgerDerivationPaths

### CODE REVIEW

File: dapp/src/utils/LedgerConnector.js [hunk 1]
```diff
@@ -66,6 +66,29 @@ export class LedgerConnector extends AbstractConnector {
     })
   }
 
+  async getTokenBalance(address, tokenAddress) {
```
Comment (sparrowDom) [inline dapp/src/utils/LedgerConnector.js:69]: with calling the `balanceOf` on contracts you can probably remove this bit.

File: dapp/src/components/LedgerDerivationContent.js [hunk 2]
```diff
@@ -38,30 +65,143 @@ const LedgerDerivationContent = ({}) => {
...
     {
-      display: `Ethereum - m/${LEDGER_OTHER}`,
+      display: `Ethereum`,
       path: LEDGER_OTHER,
     },
   ]
 
-  const onSelectDerivationPath = async (path) => {
+  const loadBalances = async (path) => {
+    if (ledgerConnector.provider && addresses[path]) {
```
Comment (sparrowDom) [inline dapp/src/components/LedgerDerivationContent.js:82]: You can early exit a function by negating a statement and avoiding `if` indentation. Changing this:  
```
if(something) {
  // a lot of lines of execution
}
```
to this:
```
if(!something) {
  return;
}
// a lot of lines of execution (not indented) 
```

**Review by sparrowDom — COMMENTED** [review]
Thanks for your work on this @rolandpo ! This is a massive Ledger connection experience improvement. Aside from a few comments inline it would be cool that we also shorten the Ledger addresses. Would make it more readable.

File: dapp/src/components/LedgerDerivationContent.js (hunk 2 above)
Comment (sparrowDom) [reply dapp/src/components/LedgerDerivationContent.js:82]: Oh another nitpick when arrow is facing upwards it is not centered: 

Screenshot 2022-03-18 at 11 18 18

**Comment by sparrowDom** [comment]
Oh another nitpick when arrow is facing upwards it is not centered:
Screenshot 2022-03-18 at 11 18 18

**Comment by rolandpo** [comment]
Appreciate your comments @sparrowDom ! I committed all the changes for those.

About the address shortening, I originally left them long to fit the wider box, but I agree that's probably too much text. I shortened the addresses, but kept the buttons wide to fill the empty space and leave room for large balances, let me know if that works.

ledgerDerivationPathsShort

**Comment by franckc** [comment]
@micahalcorn UI looks good to you? If yes, let's merge this in!

File: dapp/src/components/LedgerDerivationContent.js [hunk 3]
```diff
@@ -78,31 +206,104 @@ const LedgerDerivationContent = ({}) => {
...
-            </button>
-          )
-        })}
+        <div className={`paths d-flex flex-row`}>
+          {options.map((option) => {
+            return (
+              <button
+                key={option.path}
+                className={
+                  'text-center ' + (activePath === option.path && 'active')
```
Comment (evgenyboxer) [inline dapp/src/components/LedgerDerivationContent.js:215]: You can actually utilise `classnames` (its meant for this exactly! and is already used throughout the codebase).
Docs - https://github.com/JedWatson/classnames

File: dapp/src/components/LedgerDerivationContent.js [hunk 4]
```diff
@@ -141,6 +357,46 @@ const LedgerDerivationContent = ({}) => {
...
+          height: 25px;
+        }
+
+        .rotating {
+          -webkit-animation: spin 2s linear infinite;
+          -moz-animation: spin 2s linear infinite;
+          animation: spin 2s linear infinite;
+        }
+
+        @-moz-keyframes spin {
```
Comment (evgenyboxer) [inline dapp/src/components/LedgerDerivationContent.js:384]: Are vendor prefix needed? It should be on by default as per the docs for `styled-jsx` 
https://github.com/vercel/styled-jsx#vendorprefixes

File: dapp/src/components/LedgerDerivationContent.js [hunk 5]
```diff
@@ -38,30 +65,143 @@ const LedgerDerivationContent = ({}) => {
...
+            contractData.map((c) =>
+              ledgerConnector
+                .getTokenBalance(a, c.address)
+                .then((r) => Number(r) / 10 ** c.decimals)
+            )
+          )
+        })
+      )
+
+      let ethTotal = 0
```
[the remark this answers is not in the document]
Comment (evgenyboxer) [reply dapp/src/components/LedgerDerivationContent.js]: Yep - I'll add that you could also use lodash `sum` for better readability.

[the remark this answers is not in the document]
Comment (sparrowDom) [reply dapp/src/components/LedgerDerivationContent.js]: yup thanks that is even cleaner

**Comment by sparrowDom** [comment]
Thanks for addressing all the issues looks great @rolandpo . Great work with this PR 🙏

**Review by sparrowDom — APPROVED** [review]

**Pull request merged**